### ISSUE TITLE
Webpack compiling issue

### ISSUE THREAD

**Issue by markhomoki** [Hacker0x01/react-datepicker#956]
I was using this datepicker in the last couple months and today I updated from 0.49.0 to 0.52.0. When I start compiling with webpack it works well without any error, but when I update any js file, and webpack automatically recompiles the project I have a console error which says: 
![image](https://user-images.githubusercontent.com/8550329/28280527-e7b6ba96-6b1b-11e7-9b33-259f386bbb6c.png)

The issue is weird because when I comment out the import ReactDatePicker from 'react-datepicker'; part it works fine, and then I add it back it is still fine. But again if I do any update on my files it breaks and shows the same error in console.

Any one had similar issue?

**Comment by jcabrerazuniga** [comment]
looks like something related to the  selected  props. Is it getting a value coming from the  moment  function? (a moment object?)

**Comment by markhomoki** [comment]
This is what I thought so, but defaults to moment(). Now I created a simpler case based on the documentation and the issue is the same. I'm not sure if its related to webpack, because on first compile works fine. Then when recomplies after watch it breaks, but only because of the react-datepicker.

Here is my simplified code, which still doesn't work as it did before.

import React from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';


	export default class _DatePicker extends React.PureComponent {

		constructor (props) {
			super(props)
			this.state = {
				startDate: moment()
			};
			this.handleChange = this.handleChange.bind(this);
		}

		handleChange(date) {
			this.setState({
				startDate: date,
			});
		}

		render() {
			return (
				<DatePicker
					selected={this.state.startDate}
					onChange={this.handleChange}
				/>
			);
		}
	}

**Comment by markhomoki** [comment]
I found the solution for this issue so I am sharing if anyone has the same in the future.
I am using 2 webpack config files, 1 for production and 1 for development. The development config was running the webpack.optimize.ModuleConcatenationPlugin() plugin, which for some reason under webpack v3 crashes and does not work as expected. 

This plugin has been removed from my config file which fixed the issue. Hope this will help.
